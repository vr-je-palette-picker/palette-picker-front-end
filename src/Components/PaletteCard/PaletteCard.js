import React from 'react';
import './PaletteCard.scss'

const PaletteCard = ({ palette }) => {
  return (
    <div className='PaletteCard'>
      <p className='PaletteCard__p--palette-name'>{palette.palette_name}</p>
      <div
        className='PaletteCard__div--color'
        style={{ backgroundColor: palette.color_1 }}
      >
        {palette.color_1}
      </div>
      <div
        className='PaletteCard__div--color'
        style={{ backgroundColor: palette.color_2 }}
      >
        {palette.color_2}
      </div>
      <div
        className='PaletteCard__div--color'
        style={{ backgroundColor: palette.color_3 }}
      >
        {palette.color_3}
      </div>
      <div
        className='PaletteCard__div--color'
        style={{ backgroundColor: palette.color_4 }}
      >
        {palette.color_4}
      </div>
      <div
        className='PaletteCard__div--color'
        style={{ backgroundColor: palette.color_5 }}
      >
        {palette.color_5}
      </div>
    </div>
  );
};

export default PaletteCard;
