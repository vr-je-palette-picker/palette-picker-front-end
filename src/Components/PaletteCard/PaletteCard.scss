.PaletteCard {
  height: 27vh;
  margin: 20px;
  min-width: 200px;
}

.PaletteCard__header {
  display: flex;
  justify-content: space-between;
}

.PaletteCard__p--palette-name {
  font-size: 25px;
  margin: 0;
}

.PaletteCard__delete--icon {
  cursor: pointer;
  height: 20px;
  transform: translateY(42%);
  width: 20px;
}

.PaletteCard__div--color {
  height: 35px;
}