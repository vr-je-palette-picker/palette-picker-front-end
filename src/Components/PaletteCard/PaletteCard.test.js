import React from 'react';
import { shallow } from 'enzyme';
import PaletteCard from './PaletteCard';

describe('PaletteCard', () => {
  let wrapper;
  let palette = {
    palette_name: 'Testing 123',
    color_1: '000000',
    color_2: '00FF00',
    color_3: '999999',
    color_4: '990099',
    color_5: 'FFFFFF',
    project_id: 2
  };

  beforeEach(() => {
    wrapper = shallow(<PaletteCard palette={palette} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
