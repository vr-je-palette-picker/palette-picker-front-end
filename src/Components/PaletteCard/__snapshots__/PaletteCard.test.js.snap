// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCard should match the snapshot 1`] = `
<div
  className="PaletteCard"
>
  <p
    className="PaletteCard__p--palette-name"
  >
    Testing 123
  </p>
  <div
    className="PaletteCard__div--color"
    style={
      Object {
        "backgroundColor": "000000",
      }
    }
  >
    000000
  </div>
  <div
    className="PaletteCard__div--color"
    style={
      Object {
        "backgroundColor": "00FF00",
      }
    }
  >
    00FF00
  </div>
  <div
    className="PaletteCard__div--color"
    style={
      Object {
        "backgroundColor": "999999",
      }
    }
  >
    999999
  </div>
  <div
    className="PaletteCard__div--color"
    style={
      Object {
        "backgroundColor": "990099",
      }
    }
  >
    990099
  </div>
  <div
    className="PaletteCard__div--color"
    style={
      Object {
        "backgroundColor": "FFFFFF",
      }
    }
  >
    FFFFFF
  </div>
</div>
`;
